src/test/pow_tests.cpp

```diff
@@ -16,117 +16,81 @@ BOOST_FIXTURE_TEST_SUITE(pow_tests, BasicTestingSetup)
+    SelectParams(CBaseChainParams::MAIN);
+    const Consensus::Params& params = Params().GetConsensus();
+    
+    int64_t nLastRetargetTime = 1279296753; // NOTE: Not an actual block time
+    int64_t nThisTime = 1279297671;  // Block #68543 of Bitcoin
+    arith_uint256 bnAvg;
+    bnAvg.SetCompact(0x1c05a3f4);
+    BOOST_CHECK_EQUAL(0x1c04bceb,
+                      CalculateNextWorkRequired(bnAvg, nThisTime, nLastRetargetTime, params));
 }
 
 /* Test the constraint on the upper bound for actual time taken */
 BOOST_AUTO_TEST_CASE(get_next_work_upper_limit_actual)
 {
-    const auto chainParams = CreateChainParams(CBaseChainParams::MAIN);
-    int64_t nLastRetargetTime = 1263163443; // NOTE: Not an actual block time
-    CBlockIndex pindexLast;
-    pindexLast.nHeight = 46367;
-    pindexLast.nTime = 1269211443;  // Block #46367
-    pindexLast.nBits = 0x1c387f6f;
-    BOOST_CHECK_EQUAL(CalculateNextWorkRequired(&pindexLast, nLastRetargetTime, chainParams->GetConsensus()), 0x1d00e1fd);
+    SelectParams(CBaseChainParams::MAIN);
+    const Consensus::Params& params = Params().GetConsensus();
+    
+    int64_t nLastRetargetTime = 1269205629; // NOTE: Not an actual block time
+    int64_t nThisTime = 1269211443;  // Block #46367 of Bitcoin
+    arith_uint256 bnAvg;
+    bnAvg.SetCompact(0x1c387f6f);
+    BOOST_CHECK_EQUAL(0x1c4a93bb,
+                      CalculateNextWorkRequired(bnAvg, nThisTime, nLastRetargetTime, params));
 }
 
 BOOST_AUTO_TEST_CASE(GetBlockProofEquivalentTime_test)
 {
-    const auto chainParams = CreateChainParams(CBaseChainParams::MAIN);
+    SelectParams(CBaseChainParams::MAIN);
+    const Consensus::Params& params = Params().GetConsensus();
+    
     std::vector<CBlockIndex> blocks(10000);
     for (int i = 0; i < 10000; i++) {
-        blocks[i].pprev = i ? &blocks[i - 1] : nullptr;
+        blocks[i].pprev = i ? &blocks[i - 1] : NULL;
         blocks[i].nHeight = i;
-        blocks[i].nTime = 1269211443 + i * chainParams->GetConsensus().nPowTargetSpacing;
+        blocks[i].nTime = 1269211443 + i * params.nPowTargetSpacing;
         blocks[i].nBits = 0x207fffff; /* target 0x7fffff000... */
         blocks[i].nChainWork = i ? blocks[i - 1].nChainWork + GetBlockProof(blocks[i - 1]) : arith_uint256(0);
     }
-
+    
     for (int j = 0; j < 1000; j++) {
-        CBlockIndex *p1 = &blocks[InsecureRandRange(10000)];
-        CBlockIndex *p2 = &blocks[InsecureRandRange(10000)];
-        CBlockIndex *p3 = &blocks[InsecureRandRange(10000)];
-
-        int64_t tdiff = GetBlockProofEquivalentTime(*p1, *p2, *p3, chainParams->GetConsensus());
+        CBlockIndex *p1 = &blocks[GetRand(10000)];
+        CBlockIndex *p2 = &blocks[GetRand(10000)];
+        CBlockIndex *p3 = &blocks[GetRand(10000)];
+        
+        int64_t tdiff = GetBlockProofEquivalentTime(*p1, *p2, *p3, params);
         BOOST_CHECK_EQUAL(tdiff, p1->GetBlockTime() - p2->GetBlockTime());
     }
 }
 
```

Comment:
The current tests can't cover the logic after the fork. At least we should add them later.

Reply:
Also there are failing tests. Please make sure the unit tests and reg tests can be passed.

```
FAIL: test/test_bitcoin
=======================
Running 258 test cases...
test/pow_tests.cpp(27): error: in "pow_tests/get_next_work": check 0x1d011998 == CalculateNextWorkRequired(bnAvg, nThisTime, nLastRetargetTime, params) has failed [486611352 != 486594313]
test/pow_tests.cpp(41): error: in "pow_tests/get_next_work_pow_limit": check 0x1f07ffff == CalculateNextWorkRequired(bnAvg, nThisTime, nLastRetargetTime, params) has failed [520617983 != 486604799]
test/pow_tests.cpp(69): error: in "pow_tests/get_next_work_upper_limit_actual": check 0x1c4a93bb == CalculateNextWorkRequired(bnAvg, nThisTime, nLastRetargetTime, params) has failed [474649531 != 473066834]
```